### PR TITLE
IPO/LTO support for ICX (IntelLLVM) compiler

### PR DESCRIPTION
**Pull request by mablanchard**
## Description

Fixes https://github.com/pybind/pybind11/issues/4080

## Suggested changelog entry:

```rst
* Extend IPO/LTO detection for ICX (a.k.a IntelLLVM) compiler.
```

### CODE REVIEW

**Comment by Skylion007** [comment]
Do we test against IntelLLVM in our current CI? I am not sure we do in our OneAPI tests. If not, would it be difficult to add it as an additional workflow to our Github Actions?

**Comment by rwgk** [comment]
> Do we test against IntelLLVM in our current CI? I am not sure we do in our OneAPI tests. If not, would it be difficult to add it as an additional workflow to our Github Actions?

+1

If we don't test it we're bound to break it sooner or later, on and off.

Do we need a license for IntelLLVM?

If not it's probably pretty easy to add another section to .github/workflows/ci.yml, usually a little bit of copy-paste + trial-and-error does the trick. I've just been through something similar with #4342: deleted all existing workflows (1st commit), added a new one temporarily in a separate file for quick testing. In the case of IntelLLVM: once it's known to work copy into ci.yml.

**Review by henryiii — APPROVED** [review]
I'm not sure we do much testing of these options for the compilers, so I'd be okay to merge and then hope someone adds a CI job (and official support listed in the README) later.

**Comment by rwgk** [comment]
> I'm not sure we do much testing of these options for the compilers, so I'd be okay to merge and then hope someone adds a CI job (and official support listed in the README) later.

Sounds good to me! I went ahead and added a 1-line WARNING/HELP WANTED comment. Waiting for the CI to finish before merging (although I cannot imagine how that extra comment line could break things).

**Review by rwgk — APPROVED** [review]

**Comment by rwgk** [comment]
CI / 🐍 3 • CentOS7 / PGI 22.9 • x64 is failing:

Public key for nvhpc-2022-22.11-1.x86_64.rpm is not installed

But it worked before I added the 1-line comment:

https://github.com/pybind/pybind11/actions?query=branch%3Aicx_ipo+workflow%3Aci

I.e. the failure is clearly unrelated. Merging.

**Pull request merged**
